src/autoscaler/api/brokerserver/broker.go

```diff
@@ -0,0 +1,724 @@
+			b.logger.Error("failed to delete schedules", err, lager.Data{"appId": appId})
+		}
+	}
+	return nil
+}
+
+func (b *Broker) checkScalingPoliciesUnderNewPlan(allBoundApps []string, servicePlan string, instanceID string) error {
+	var existingPolicy *models.ScalingPolicy
+	var existingPolicyByteArray []byte
+	var err error
+	for _, appId := range allBoundApps {
+		existingPolicy, err = b.policydb.GetAppPolicy(appId)
+		if err != nil {
+			b.logger.Error("failed to retrieve policy from db", err, lager.Data{"appId": appId})
+			return apiresponses.NewFailureResponse(errorUpdatingServiceInstance, http.StatusInternalServerError, "failed to retrieve policy from db")
+		}
+		existingPolicyByteArray, err = json.Marshal(existingPolicy)
+		if err != nil {
+			b.logger.Error("failed to marshal policy from db", err, lager.Data{"appId": appId})
+			return apiresponses.NewFailureResponse(errorUpdatingServiceInstance, http.StatusInternalServerError, "failed to marshal policy from db")
+		}
+		existingPolicyStr := string(existingPolicyByteArray)
+		if err := b.planDefinitionExceeded(existingPolicyStr, servicePlan, instanceID); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (b *Broker) determineDefaultPolicy(parameters *models.InstanceParameters, serviceInstance *models.ServiceInstance, planID string) (string, string, bool, error) {
+	defaultPolicy := serviceInstance.DefaultPolicy
+	defaultPolicyGuid := serviceInstance.DefaultPolicyGuid
+	defaultPolicyIsNew := false
+	var err error
+
+	if parameters.DefaultPolicy == nil {
+		return defaultPolicy, defaultPolicyGuid, false, nil
+	}
+
+	newDefaultPolicy := string(*parameters.DefaultPolicy)
+	if emptyJSONObject.MatchString(newDefaultPolicy) {
+		// accept an empty json object "{}" as a default policy update to specify the removal of the default policy
+		if defaultPolicy != "" {
+			defaultPolicy = ""
+			defaultPolicyGuid = ""
+			defaultPolicyIsNew = true
+		}
+	} else {
+		if newDefaultPolicy != defaultPolicy {
+			newDefaultPolicy, err = b.validateAndCheckPolicy(newDefaultPolicy, serviceInstance.ServiceInstanceId, planID)
+			if err != nil {
+				return "", "", false, err
+			}
+
+			policyGuid, err := uuid.NewV4()
+			if err != nil {
+				b.logger.Error("determine-default-policy-create-guid", err, lager.Data{"instanceID": serviceInstance.ServiceInstanceId})
+				return "", "", false, apiresponses.NewFailureResponse(errors.New("failed to create policy guid"), http.StatusInternalServerError, "determine-default-policy-create-guidz")
+			}
+			defaultPolicy = newDefaultPolicy
+			defaultPolicyGuid = policyGuid.String()
+			defaultPolicyIsNew = true
+		}
+	}
+
+	return defaultPolicy, defaultPolicyGuid, defaultPolicyIsNew, nil
+}
+
+// LastOperation fetches last operation state for a service instance
+// GET /v2/service_instances/{instance_id}/last_operation
+func (b *Broker) LastOperation(_ context.Context, instanceID string, details domain.PollDetails) (domain.LastOperation, error) {
```

Comment:
hmm should we add a test?

Reply:
I think there are a few like this that could share the same NotImplemented function

Reply:
I'll leave that open for now, as this whole refactoring is just a prelude to actually implementing this methods.